src/include/sof/audio/component.h

```diff
@@ -145,23 +145,34 @@ struct dai_hw_params;
 /** \name Trace macros
  *  @{
  */
-#define trace_comp(__e, ...) \
+#define trace_comp_get_id(comp_p) ((comp_p)->comp.pipeline_id)
+#define trace_comp_get_subid(comp_p) ((comp_p)->comp.id)
+
+/* class (driver) level (no device object) tracing */
+
+#define comp_cl_err(drv_p, __e, ...)					\
+	trace_error(TRACE_CLASS_COMP, __e, ##__VA_ARGS__)
```

Comment:
I have noticed you have done the first step towards getting proper trace classes for the comp_cl_* macros, now I think the class can in fact be stored in drv_p and you should pull it from there rather than using TRACE_CLASS_COMP.

Reply:
Not the class, but a ptr to static module info descriptor put into the ldc file. I'd keep the class for base parts of the fw while use second level identification (uuid, not anything merged as a flat list) for components and dais. Will explain that in the final description once compilation issues are fixed soon.

Reply:
Yeah fair, but the class must still be there (not willing to accept breaking backward compatibility in multiple pull requests and this one is complex enough as-is so I want it preserved), at least for now. Changing the format of the trace message I recommend you do in a follow-up.

Reply:
I updated the description and title. Builds and tests are passing. I had to add protection to trace header and final solution would require more serious cleanup but this is another story for another patch (look at the order of using pre-prod directives and other includes). I tried to explain why I do not want to move trace classes to the context, while adding uuids identification to id later (two different use cases, doubt if the former is really worth the overhead). I also tried to explain next steps for uuids and in the meantime users may still easily filter their logs if they need) and I would not restore the old trace classes for components and dais temporarily unless somebody raises serious concern and real issue with that.

Reply:
Fair enough.